File: Threads/src/Java/Kuehlschrank.java
package Java;

import java.util.ArrayList;
import java.util.List;

public class Kuehlschrank {
    private final List<Yogurt> yogurt;

    public Kuehlschrank() {
        this.yogurt = new ArrayList<>();
    }

    public String hinfzufuegen(Yogurt ding) {
        yogurt.add(ding);
        return "hat Yogurt " + ding.getName() + " hinzugefügt.";
    }

    public String entnehmen() {
        return "hat Yogurt " + yogurt.remove(yogurt.size() - 1).getName() + " entnommen.";
    }

    public int yogurtAnzahl() {
        return yogurt.size();
    }

    public boolean istLeer() {
        return yogurt.isEmpty();
    }

    public boolean nochPlatz() {
        return yogurt.size() < 10;
    }
}

File: Threads/src/Kuehlschrank.kt
class Kuehlschrank() {

    private val yogurt: MutableList<Yogurt> = mutableListOf()

    fun hinfzufuegen(ding: Yogurt) {
        if (yogurt.size < 10)
            yogurt.add(ding)
    }

    fun entnehmen() {
        if (yogurt.size > 0)
            yogurt.removeAt(yogurt.size - 1)
    }

    fun yogurtAnzahl() = yogurt.size

    fun nochPlatz():Boolean = yogurt.size < 10

}

class Yogurt(private val name: String) {

}